test/conftest.py

```diff
@@ -24,39 +23,38 @@ def multisensor_6_state_fixture():
     return json.loads(load_fixture("multisensor_6_state.json"))
 
 
-@pytest.fixture(name="client_session")
-def client_session_fixture():
-    """Mock an aiohttp client session."""
-    return AsyncMock(spec_set=ClientSession)
+class MockClient(Client):
+    """Mock client."""
 
+    def __init__(self):
+        """Initialize mock client."""
+        super().__init__("ws://test.org", None)
+        self.state = STATE_CONNECTED
+        self.mock_responses: List[Tuple[dict, dict]] = []
+        self.mock_commands = []
 
-@pytest.fixture(name="ws_client")
-def ws_client_fixture():
-    """Mock a websocket client."""
-    return AsyncMock(spec_set=ClientWebSocketResponse)
+    async def async_send_command(self, message):
```

Comment:
I don't like that we're changing internal things on the client. We should patch in the periphery instead, ie the transport.

Reply:
I disagree. Mocking the transport is only needed when you're going to test the actual client. When we're testing our models, we just need a stand-in client.

Reply:
If we change the client we'll need to update the tests. I don't like this. Patching in the periphery makes sure we minimize the dependency of the tests on our implementation.

Reply:
We would also need to update the server and the code that calls it. That seems like things that should change together.

Reply:
If we patch the transport we'll only need to change the tests if we change transport. That's a lot less likely to happen than that we change how we use the transport in the client. So all the tests should be more stable in general that way.

Another benefit is that the mocks are more predictable since we always patch in the same place and the interface for that will look the same, regardless of what part of the library we're testing.

Reply:
I think that it's a premature optimization to make the architecture flexible enough in case we are going to change the way we're going to send commands. If you feel like it's a showstopper for getting basic command stubbing in the test suite, feel free to close this PR.

Reply:
I'll open a PR with my suggestion and we can decide then.